Customize /find/accounts and "No Organization Found" Pages for Self-Hosted Servers.

shuklamaneesh23 opened the pull request:
This PR customizes the `/find/accounts` page and the "No Organization Found" page for self-hosted Zulip servers. Previously, these pages were tailored for Zulip Cloud organizations, but this update ensures that self-hosted servers display more appropriate messages.

### Changes:
1. **Customize `/find/accounts` page for self-hosted servers**:
   - The message displayed is now: 
     - For Zulip Cloud: "for all the Zulip Cloud organizations in which you have active accounts."
     - For self-hosted servers: "for all the Zulip organizations on this server."
   - This change makes the text more relevant to users on self-hosted instances.

2. **Update the "No Organization Found" page**:
   - For Zulip Cloud:
     - Message updated to: "Please try a different URL, get a list of your Zulip Cloud accounts, or contact..."
   - For self-hosted servers:
     - Message updated to: "Please try a different URL, get a list of your accounts on this server, or contact..."

Fixes: #30116 

---

### Previous PR Reference:
A previous PR #30207 addressed the same issue but has been inactive since June. This PR builds on the foundational work done in that earlier PR, which similarly focused on updating frontend text and instructions for self-hosted users.

While the earlier PR made valuable progress, there were a few gaps that this PR addresses:
- **Commit messages**: The previous PR's commit messages did not fully align with the project's contribution guidelines.
- **Incorrect link**: The link added to the "No Organization Found" page was incorrect in the earlier implementation.

This PR not only finalizes the original fixes but also ensures that all changes, including the commit history, follow the required standards. The rest of the frontend code remains consistent with the previous work, making this a continuation and completion of that effort.

---
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details> 

<summary> Current UI: </summary>
<img width="728" alt="Screenshot 2024-10-08 at 8 37 40 PM" src="https://github.com/user-attachments/assets/4bf396bf-db7b-4710-9303-2accd5f87e80">
<img width="485" alt="Screenshot 2024-10-08 at 8 41 21 PM" src="https://github.com/user-attachments/assets/bb085f53-af08-4d80-88a3-a025fae833d7">


</details>

<details> 

<summary> Updated UI for Zulip Cloud: </summary>
<img width="728" alt="Screenshot 2024-10-08 at 8 37 40 PM" src="https://github.com/user-attachments/assets/9a8abae8-c268-493f-8fbf-67075cd55252">
<img width="1440" alt="Screenshot 2024-10-22 at 10 31 05 AM" src="https://github.com/user-attachments/assets/ad4f1a88-ee5c-466d-96da-034a336a7fb8">



</details>

<details> 

<summary> Updated UI for Self-Hosted servers: </summary>
<img width="690" alt="Screenshot 2024-10-09 at 2 49 53 PM" src="https://github.com/user-attachments/assets/4be1345b-4eb2-4577-8065-49ebaa3f1cf5">
<img width="1440" alt="Screenshot 2024-10-22 at 10 32 38 AM" src="https://github.com/user-attachments/assets/53cb61b3-1c8a-4ab2-8f26-12a7cbeccd54">





</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

shuklamaneesh23 commented:
Hey @alya ,can you please review this?

alya commented:
Thanks! Let's limit the width of the text box to be the same as the one on the "Find your Zulip accounts" page.

shuklamaneesh23 commented:
> Thanks! Let's limit the width of the text box to be the same as the one on the "Find your Zulip account
<img width="810" alt="Screenshot 2024-10-16 at 2 10 41 PM" src="https://github.com/user-attachments/assets/a565171b-9a77-4d85-b906-2240a830a73e">
s" page.

If I am understanding it clearly, are you talking about this?
Should I put the content of the second line to a new line?

shuklamaneesh23 commented:
> Thanks! Let's limit the width of the text box to be the same as the one on the "Find your Zulip accounts" page.

Hey @alya  , If I am understanding it clearly, are you talking about below page?
Should I put the content of the second line to a new line?
<img width="800" alt="Screenshot 2024-10-17 at 4 17 12 AM" src="https://github.com/user-attachments/assets/57a0379c-a751-4bf5-87e8-80f829d37db4">

alya commented:
The content on the second line will wrap when you limit the width, yes.

shuklamaneesh23 commented:
Hey @alya , updated the PR description with screenshots of updated "No Organisation Found" Page.
Could you please review it again.
Thanks!

alya commented:
Let's add a line break between the paragraphs.

shuklamaneesh23 commented:
Hey @alya , added the line break in the paragraph.
Thanks!

shuklamaneesh23 commented:
Hello @alya , can you please review this up?
Thanks!

shuklamaneesh23 commented:
Hey @alya , could you please check this up?

shuklamaneesh23 commented:
Hey @alya , could you please check this up?

alya commented:
@laurynmm could you please do the first round of review on this one?

laurynmm reviewed (commented):
@shuklamaneesh23 - Thanks for working on these updates! Left a few review comments. Let me know if you have any questions about that feedback.

Also, you'll need to clean up your [commit messages](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages) for a few things:
- There's a typo in the commit summary for both commits.
- In the first commit:
  - There's no need to note the text change in the commit description.
  - Instead you can just describe what the changes do.
- In the second commit:
  - Your changes don't actually update the page to distinguish between Zulip Cloud and self-hosted organizations.
  - Your changes expand the text to have a link to the "find accounts" page, and add the "find-account-page-container" class so that the width of the white box with text on the page is limited.
  - Not having it noted in the commit message why that class change happened meant that I had to read through the comments on the PR in GitHub to understand why it was there.

I linked to our documentation on commit messages, but if you have any questions about my notes above, let me know!

shuklamaneesh23 commented:
Hey @laurynmm,

I've addressed all the concerns raised in your feedback:
- Corrected the typos in the commit summaries.
- Updated the commit descriptions to focus on what the changes do, without listing the exact text changes.
- Clarified the purpose of the "find-account-page-container" class change and its impact on the layout in the second commit message.

Could you please review it again when you get a chance?

Thanks for your help!

laurynmm commented:
@shuklamaneesh23 - Thanks for making those updates! The code changes look good.

There was still a typo in the commit summaries (protico -> portico), so I've pushed updates to the commit messages, so that we can mark this as ready for integration review.

1st commit:
```
portico: Customise "Find your accounts" page.

Updates "Find your accounts" page to display a relevant message for
self-hosted servers, with no changes to the Zulip Cloud version of
the page.
```

2nd commit:
```
portico: Update "No organization found" page.

Expands the "No organization found" page to include a link to the
"Find your accounts" page.

Also adjusts the layout of the page for the longer text by adding
the "find-account-page-container" class for the textbox width.

Fixes #30116.
```

shuklamaneesh23 commented:
Thanks @laurynmm for the correction.

timabbott commented:
Merged, thanks @shuklamaneesh23!
